src/Parallel/CMakeLists.txt

```diff
@@ -52,6 +52,7 @@ target_link_libraries(
   INTERFACE
   Boost::boost
   DataStructures
+  EventsAndTriggers
```

Comment:
Note that this introduces a cyclic dependency (of course it was there before, this change only makes it explicit). Perhaps it can be resolved by making `ParallelPhaseControl` a separate library, though I'm note sure since it's used in `Parallel/Main.hpp`. Maybe we need a separate `ParallelMain` lib as well. What do you think @moxcodes?

Reply:
@moxcodes do you have input here?

Reply:
EventsAndTriggers depends on Parallel, but doesn't currently list it in `targt_link_libraries`. Note all the Parallel/... includes. Also, all libs in `ParallelAlgorithms` depend on Parallel.

Reply:
Okay, so the parts that the events and triggers need are the cache and the serialization utiltities, and the parts that need triggers are the phase changes and anything that uses them.
It certainly makes sense to split out `PhaseControl` to its own lib, I'm happy with that. Splitting out main seems more annoying. I think all that main depends on directly is the tags, so would it make sense to move the `PhaseControlTags.hpp` up to the `Parallel` directory, include it in `Parallel` lib (I think the include of triggers there needs to be converted to a forward decl, but that appears fine in that file), then `PhaseControl` can be its own lib that depends on triggers, and wouldn't appear to have a cyclic dependency anymore.

Reply:
I think merging this PR and fixing the cyclic dependency separately will be fine 👍 I created https://github.com/sxs-collaboration/spectre/issues/3163. Thanks @moxcodes .